addon/lint/lint.js

```diff
@@ -187,7 +187,7 @@
   CodeMirror.defineOption("lint", false, function(cm, val, old) {
     if (old && old != CodeMirror.Init) {
       clearMarks(cm);
-      if (!state.options.lintOnChange !== false)
+      if (cm.state.lint.options.lintOnChange === undefined || cm.state.lint.options.lintOnChange === true)
```

Comment:
COND = !state.options.lintOnChange !== false
truth table:
lintOnChange: undefined. Should lint on change? YES. COND is true. OK.
lintOnChange: true. Should lint on change? YES. COND is false. ERR.
lintOnChange: false. Should lint on change? NO. COND is true. ERR.

Am I right?